### DISCUSSION TITLE
How to enforce that it is only startup/program that has a reference to infrastructure?

### DISCUSSION THREAD

**Discussion by ivnie** [jasontaylordev/CleanArchitecture#731] [Ideas]
From the blog:
_The WebUI project represents the Presentation layer. This project is a SPA (single page app) based on Angular 8 and ASP.NET Core. This layer depends on both the Application and Infrastructure layers. Please note the dependency on Infrastructure is only to support dependency injection. Therefore Startup.cs should include the only reference to Infrastructure._

Is there a good way to enforce that it is only startup/program that has a reference to infrastructure?
I was thinking something like using code analyzers to fail build if there are references to Infrastructure outside startup/program, but is this even a good approach?

**Comment by neman** [comment]
You could unit test architecture with this https://archunitnet.readthedocs.io/en/latest/ and prevent other layers to reference infrastructure.
